File: src/classes/status/status.rs
use super::{status_hvac::StatusHVAC, status_light::StatusLight};

pub struct Status{
	status_type: u8,
	light: Option<StatusLight>,
	hvac: Option<StatusHVAC>,
	//curtain: Option<StatusCurtain>,
	//ias_sensor: Option<StatusIAS>,
	//door_lock: Option<StatusDoorLock>,
}

impl Status{
	
}
/*
public class HACommand_Status
{
	public EndpointValues pTheEndpointValues;

	protected byte StatusType { get; private set; }

	private HACommand_StatusLight Light { get; set; }

	private HACommand_StatusHVAC HVAC { get; set; }

	//private HACommand_StatusCurtain Curtain { get; set; }

	//private HACommand_StatusIAS IASSensor { get; set; }

	//private HACommand_StatusDoorLock DoorLock { get; set; }

	public EndpointValues TheEndpointValues
	{
		get
		{
			switch (StatusType)
			{
			case 4:
				//pTheEndpointValues.Values = Curtain.EndpointValues;
				break;
			case 6:
				pTheEndpointValues.Values = HVAC.EndpointValues;
				break;
			case 1:
				pTheEndpointValues.Values = Light.EndpointValues;
				break;
			case 8:
				//pTheEndpointValues.Values = IASSensor.EndpointValues;
				break;
			case 7:
				//pTheEndpointValues.Values = DoorLock.EndpointValues;
				break;
			}
			return pTheEndpointValues;
		}
	}

	public HACommand_Status()
	{
		pTheEndpointValues = new EndpointValues();
	}

	public void SetBytes(byte[] data, int headerOffset)
	{
		StatusType = data[headerOffset];
		switch (StatusType)
		{
		case 4:
			/*if (Curtain == null)
			{
				Curtain = new HACommand_StatusCurtain();
			}
			Curtain.SetBytes(data, headerOffset + 1);*/
			break;
		case 6:
			if (HVAC == null)
			{
				HVAC = new HACommand_StatusHVAC();
			}
			HVAC.SetBytes(data, headerOffset + 1);
			break;
		case 1:
			if (Light == null)
			{
				Light = new HACommand_StatusLight();
			}
			Light.SetBytes(data, headerOffset + 1);
			break;
		case 8:
			/*if (IASSensor == null)
			{
				IASSensor = new HACommand_StatusIAS();
			}
			IASSensor.SetBytes(data, headerOffset + 1);*/
			break;
		case 7:
			/*if (DoorLock == null)
			{
				DoorLock = new HACommand_StatusDoorLock();
			}
			DoorLock.SetBytes(data, headerOffset + 1);*/
			break;
		case 2:
		case 3:
		case 5:
			break;
		}
	}
}

*/

File: lib/classes/status/status.py
from lib.classes.status.status_hvac import StatusHVAC
from lib.classes.status.status_light import StatusLight


class Status:
    def __init__(self):
        self.endpoint_values = []
        self.status_type = 0
        self.light = None
        self.hvac = None
        #self.curtain = None
        #self.ias_sensor = None
        #self.door_lock = None

    def update_status(self, status_type):
        self.status_type = status_type
        self.endpoint_values = self.match_endpoint_values()

    def match_endpoint_values(self):
        if self.status_type == 1:
            if self.light is None:
                self.light = StatusLight()
            return self.light.get_endpoint_values()
        elif self.status_type == 6:
            if self.hvac is None:
                self.hvac = StatusHVAC()
            return self.hvac.get_endpoint_values()
        return []
    def set_bytes(self, data, header_offset):
        self.status_type = data[header_offset]
        if self.status_type == 6:
            if self.hvac is None:
                self.hvac = StatusHVAC()
            self.hvac.set_bytes(data, header_offset + 1)
        elif self.status_type == 1:
            if self.light is None:
                self.light = StatusLight()
            self.light.set_bytes(data, header_offset + 1)
        return True

    def get_endpoint_values(self):
        return self.endpoint_values

    def get_status_type(self):
        return self.status_type

    def get_light(self):
        return self.light

    def set_light(self, light):
        self.light = light

    def get_hvac(self):
        return self.hvac

    def set_hvac(self, hvac):
        self.hvac = hvac

    """
    def get_curtain(self):
        return self.curtain

    def set_curtain(self, curtain):
        self.curtain = curtain

    def get_ias_sensor(self):
        return self.ias_sensor

    def set_ias_sensor(self, ias_sensor):
        self.ias_sensor = ias_sensor

    def get_door_lock(self):
        return self.door_lock

    def set_door_lock(self, door_lock):
        self.door_lock = door_lock
    """
    
